src/ngraph/frontend/onnx_import/op/eye_like.cpp

```diff
@@ -0,0 +1,87 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <onnx-ml.pb.h>
+
+#include "exceptions.hpp"
+#include "eye_like.hpp"
+#include "ngraph/builder/reshape.hpp"
+#include "ngraph/frontend/onnx_import/utils/common.hpp"
+#include "ngraph/op/argmax.hpp"
+#include "ngraph/op/embedding_lookup.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace set_1
+            {
+                NodeVector eye_like(const Node& node)
+                {
+                    const auto input = node.get_ng_inputs().at(0);
+                    const auto& input_shape = input->get_shape();
+
+                    std::int64_t dtype;
+                    element::Type target_type;
+
+                    std::int64_t shift = node.get_attribute_value<std::int64_t>("k", 0);
+                    try
+                    {
+                        dtype = node.get_attribute_value<std::int64_t>("dtype");
+                        switch (dtype)
+                        {
+                        case onnx::TensorProto_DataType_BOOL: target_type = element::boolean; break;
+                        case onnx::TensorProto_DataType_DOUBLE: target_type = element::f64; break;
+                        case onnx::TensorProto_DataType_FLOAT16:
```

Comment:
ngraph has f16, use it here.

Reply:
I added support for `fp16` but when I'm trying to use `Convert` op with `fp16` type I got `bad function call` and `Cannot convert from an invalid input element type` on `CPU` backend.
`INTERPRETER` backend also don't support this type: `unsupported element type element::Type{16, 1, 1, 0, "float16"} op Convert`.